icu4c/source/common/unicode/brkiter.h

```diff
@@ -93,6 +95,34 @@ U_NAMESPACE_BEGIN
  * are available at http://www.unicode.org/reports/tr14/ and
  * http://www.unicode.org/reports/tr29/.
  * <p>
+ * A boundary logically occurs between two adjacent code points. The boundary
+ * analysis functions in this file return the index of the code unit directly
+ * after a boundary - so a value of 0 means "the beginning of the string" and a
+ * value of u_strlen(string) means "the end of the string".
+ * <p>
+ * A span of text between two adjacent boundaries is called a "segment."
+ * The boundary analysis functions in this file behave correctly regardless of
+ * whether the current position is set to a boundary location or into the
+ * middle of a segment. When the current position points to a boundary, the two
+ * segments that straddle that boundary are called the "previous segment" and
+ * the "current segment."
+ * <p>
+ * If the current position is set to the middle of a surrogate pair, the
+ * boundary analysis functions in this file will move the current position to
+ * the beginning of the surrogate pair before any other processing. If the
+ * current position then points into the middle of a segment, these routines
+ * will move to the beginning or end of that segment. Otherwise, the current
```

Comment:
"will move to the beginning or end of that segment."
how about change to
"will move to segment boundary before or after that segment."
to make it consistent w/ the following sentance.

Reply:
Again, an ICU break iterators's current position is fully constrained to always be on a boundary. So neither of the proposed changes to this text works.